algorithms/src/fft/polynomial/mod.rs

```diff
@@ -67,7 +67,12 @@ impl<'a, F: Field> CanonicalSerialize for Polynomial<'a, F> {
 
 impl<'a, F: Field> Valid for Polynomial<'a, F> {
     fn check(&self) -> Result<(), SerializationError> {
-        Ok(())
+        // Fail if the trailing coefficient is zero
+        if let Some(false) = self.coeffs().last().map(|c| c.1.is_zero()) {
```

Comment:
```suggestion
        if let Some(true) = self.coeffs().last().map(|c| c.1.is_zero()) {
```
unless it's the bad sleep getting to me, shouldn't this be `true` instead?

Reply:
You are very right. Thank you for spotting.